Services/EnergyCostAlgorithm.cs

```diff
@@ -0,0 +1,30 @@
+
+namespace Ultra_Saver;
+
+public class EnergyCostAlgorithm
+{
+    private static double totalEnergy = 0;
+    private static double totalEnergyCost = 0;
+    private static float costPerkWh = 0;
```

Comment:
Not sure if these variables should be static, we might have problems if there will be multiple instances of this object (not sure though)
Also, look into properties, further explanation below

Reply:
This probably is going to create problems